Draft bap thread introduction.

rasmushoeg opened the pull request:
Changes included:

- Added bap_thread() and bap_sink_reset() in bap_broad_snk.c - from line 740. 
- Introduced bap_unicast_sr_has_connection() as an API to encapsulate access to default_conn.
- removed self-scan logic in bap_thread.

rasmushoeg committed:
hap_ha: initial broadcast_sink

This PR adds the initial base functionality of the broadcast sink.
The functionality is test on a NRF5340 audio.
Minor fixes from the latest PR.

Signed-off-by: Rasmus Moeller <rslm@demant.com>

rasmushoeg committed:
hap_ha: initial broadcast_sink

This PR adds the initial base functionality of the broadcast sink.
The functionality is test on a NRF5340 audio.

Signed-off-by: Rasmus Moeller <rslm@demant.com>

rasmushoeg committed:
hap_ha: initial broadcast_sink

This PR adds the initial base functionality of the broadcast sink.
The functionality is test on a NRF5340 audio.

Signed-off-by: Rasmus Moeller <rslm@demant.com>

rasmushoeg committed:
Added bap_thread for feedback

Added bap_thread in broadcast_snk.c for feedback while working on it.
Draft to be deleted once feedback is incorperated in the
add_bass_to_hap_ha branch.

rasmushoeg committed:
Added bap_thread for feedback

Added bap_thread in broadcast_snk.c for feedback while working on it.
Draft to be deleted once feedback is incorperated in the
add_bass_to_hap_ha branch.

rasmushoeg committed:
Added bap_thread for feedback

Added bap_thread in broadcast_snk.c for feedback while working on it.
Draft to be deleted once feedback is incorperated in the
add_bass_to_hap_ha branch.

fredrikdanebjer commented on samples/bluetooth/hap_ha/src/bap_broadcast_snk.c:
You dont need this assignment since `state` will always be `BAP_STATE_WAIT_BA` if you are inside this if statement.

fredrikdanebjer commented on samples/bluetooth/hap_ha/src/bap_broadcast_snk.c:
Shouldn't the state be set to `BAP_STATE_RESET` here? The if-statement on line 850 checks whether we `k_sem_take` is a success, and if so we put state into  `BAP_STATE_PA_SYNC` and break - so here I assume we have a failare and should reset?

fredrikdanebjer commented on samples/bluetooth/hap_ha/src/bap_broadcast_snk.c:
This function is also called "init_bap_sink" - and I know you didnt name this. But since this file now lives in a sample where both unicast and broadcast coexist, I would consider renaming it to better reflect that this refers to Broadcast, instead of relying on it being implied.

This might be a reoccurring "problem" here :)

rasmushoeg committed:
Updated PR draft.

This commit adds the suggestions changes to the code.
This is still a draft that is waiting that needs internal review.

Signed-off-by: Rasmus Moeller <rslm@demant.com>
